pkg/disk/partition_table.go

```diff
@@ -36,28 +36,34 @@ const (
 	// RawPartitioningMode always creates a raw layout.
 	RawPartitioningMode PartitioningMode = "raw"
 
+	// BtrfsPartitioningMode creates a btrfs layout.
+	BtfrsPartitioningMode PartitioningMode = "btrfs"
+
 	// DefaultPartitioningMode is AutoLVMPartitioningMode and is the empty state
 	DefaultPartitioningMode PartitioningMode = ""
 )
 
 func NewPartitionTable(basePT *PartitionTable, mountpoints []blueprint.FilesystemCustomization, imageSize uint64, mode PartitioningMode, requiredSizes map[string]uint64, rng *rand.Rand) (*PartitionTable, error) {
 	newPT := basePT.Clone().(*PartitionTable)
 
-	if basePT.features().LVM && mode == RawPartitioningMode {
-		return nil, fmt.Errorf("raw partitioning mode set for a base partition table with LVM, this is unsupported")
+	if basePT.features().LVM && (mode == RawPartitioningMode || mode == BtfrsPartitioningMode) {
+		return nil, fmt.Errorf("%s partitioning mode set for a base partition table with LVM, this is unsupported", mode)
 	}
 
 	// first pass: enlarge existing mountpoints and collect new ones
 	newMountpoints, _ := newPT.applyCustomization(mountpoints, false)
 
 	var ensureLVM bool
+	ensureBtrfs := false
```

Comment:
(super nitpick) maybe define in the same way as ensureLVM in the line above (i.e. jus t`var ensureLVM, ensureBtrfs bool`).